sourcemap/fleet_store.go

```diff
@@ -0,0 +1,140 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package sourcemap
+
+import (
+	"bytes"
+	"compress/zlib"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+
+	"github.com/elastic/apm-server/beater/config"
+	logs "github.com/elastic/apm-server/log"
+)
+
+type fleetStore struct {
+	apikey    string
+	c         *http.Client
+	fleetURLs map[key]string
+}
+
+type key struct {
+	ServiceName    string
+	ServiceVersion string
+	BundleFilepath string
+}
+
+// NewFleetStore returns an instance of Store for interacting with sourcemaps
+// stored in Fleet-Server.
+func NewFleetStore(
+	c *http.Client,
+	fleetCfg *config.Fleet,
+	cfgs []config.SourceMapMetadata,
+	expiration time.Duration,
+) (*Store, error) {
+	if len(fleetCfg.Hosts) < 1 {
+		return nil, errors.New("no fleet hosts present for fleet store")
+	}
+	logger := logp.NewLogger(logs.Sourcemap)
+	s, err := newFleetStore(c, fleetCfg, cfgs)
+	if err != nil {
+		return nil, err
+	}
+	return newStore(s, logger, expiration)
+}
+
+func newFleetStore(
+	c *http.Client,
+	fleetCfg *config.Fleet,
+	cfgs []config.SourceMapMetadata,
+) (fleetStore, error) {
+	// TODO(stn): This is assuming we'll get a single fleet address right now.
```

Comment:
can you please update the comment to resolve the question? maybe link to the issue